test/jdk/valhalla/valuetypes/IsIdentityClassTest.java

```diff
@@ -44,48 +41,51 @@
 import java.util.List;
 import java.util.Set;
 
+import jdk.internal.misc.PreviewFeatures;
 import jdk.internal.value.ValueClass;
 
 import static jdk.test.lib.Asserts.*;
 
 public class IsIdentityClassTest {
 
-    static boolean UseArrayFlattening = false;
-    static {
+    private static void assertFalseIfPreview(boolean condition, String msg) {
+        if (PreviewFeatures.isEnabled()) {
```

Comment:
We can use `assertEquals(!PreviewFeatures.isEnabled(), msg)`

Reply:
Thats a good suggestion.